### PR TITLE
Create a lru bitmap cache for loading resource drawables

### PR DESCRIPTION
**Pull request by yiqunw700**
Towards #1556

### CODE REVIEW

File: designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    fun loadResource(resources: Resources, resId: Int): Bitmap? {
+        val imageKey: String = resId.toString()
+
+        val bitmap: Bitmap? =
+            getBitmapFromMemCache(imageKey)
+                ?: run {
+                    val resBitmap = BitmapFactory.decodeResource(resources, resId)
+                    if (resBitmap != null) {
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt:48]: Done.

File: designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        putBitmapToMemCache(imageKey, resBitmap)
+                    }
+                    resBitmap
+                }
+        return bitmap
+    }
+
+    private fun getBitmapFromMemCache(key: String?): Bitmap? {
+        if (key != null) {
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt]: Done.

File: designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import android.graphics.BitmapFactory
+import androidx.collection.LruCache
+
+/** https://developer.android.com/topic/performance/graphics/cache-bitmap */
+object BitmapFactoryWithCache {
+    private val bitmapCache: LruCache<String, Bitmap>
+
+    init {
+        val maxMemory = (Runtime.getRuntime().maxMemory() / 1024).toInt()
+        val cacheSize = maxMemory / 8
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt:30]: It is the max size of the lru cache.
 `On a normal/hdpi device this is a minimum of around 4MB (32/8).` from https://developer.android.com/topic/performance/graphics/cache-bitmap
It differs on different devices. On Seahawk, it is 65536 kb(~64M).

[the remark this answers is not in the document]
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/BitmapFactoryWithCache.kt:30]: I saw that, and that seems really small. A device only has 32MB memory for the app? Seems outdated to me, but I could be wrong. Ok, I just want to make a note that we may want to make this configurable in the future, as different OEMs could be using vastly different hardware profiles with different amount of memory, and they may want to have some control over the amount of memory this uses. Can you make a low priority ticket to track this?

**Review by rylin8 — APPROVED** [review]

**Pull request merged**